processing/src/main/java/io/druid/segment/data/CompressedLongsIndexedSupplier.java

```diff
@@ -214,6 +216,65 @@ public void remove()
     );
   }
 
+  public static CompressedLongsIndexedSupplier fromList(
+      final List<Long> list , final int chunkFactor, final ByteOrder byteOrder, CompressedObjectStrategy.CompressionStrategy compression
+  )
+  {
+    Preconditions.checkArgument(
+        chunkFactor <= MAX_LONGS_IN_BUFFER, "Chunks must be <= 64k bytes. chunkFactor was[%s]", chunkFactor
+    );
+
+    return new CompressedLongsIndexedSupplier(
+        list.size(),
+        chunkFactor,
+        GenericIndexed.fromIterable(
+            new Iterable<ResourceHolder<LongBuffer>>()
+            {
+              @Override
+              public Iterator<ResourceHolder<LongBuffer>> iterator()
+              {
+                return new Iterator<ResourceHolder<LongBuffer>>()
+                {
+                  int position = 0;
```

Comment:
it seems a portion of this is copy pasta. Is it possible to commonalize some of the logic?